data/scripts/sol-oic-gen.py

```diff
@@ -1220,7 +1220,7 @@ def master_c_as_string(generated, oic_gen_c, oic_gen_h):
     }
 
     SOL_INF("Found resource matching device_id");
-    resource->resource = sol_oic_resource_ref(oic_res);
+    resource->resource = sol_oic_resource_ref((struct sol_oic_resource *)oic_res);
```

Comment:
I just saw that you made refcount private. Is it a problem to change make sol_oic_resource_ref  to receive a non-const sol_oic_resource?